tally/statgram.go

```diff
@@ -184,6 +189,17 @@ func ParseSample(key string, part []byte) (sample Sample, err error) {
 
 var stringSampleReplacer *strings.Replacer
 
+func decodeGaugeSample(sample []byte) (value float64, replace bool) {
+	replace = false
+	value, _ = ParseFloat(sample)
+
+	if _, err := strconv.Atoi(string(sample[0])); err == nil {
```

Comment:
Looking at this further, this is icking me out a bit: is this trying to figure out if the first character is a + or - by seeing if that character by itself can be turned into an integer?